src/main/java/com/minecolonies/event/EventHandler.java

```diff
@@ -29,6 +33,45 @@
 public class EventHandler
 {
     /**
+     * Event when the debug screen is opened.
+     * Event gets called by displayed text on the screen, we only need it when f3 is clicked.
+     *
+     *  @param event {@link net.minecraftforge.client.event.RenderGameOverlayEvent.Text}
+     */
+    @SubscribeEvent
+    public void onDebugOverlay(RenderGameOverlayEvent.Text event)
+    {
+        final Minecraft mc = Minecraft.getMinecraft();
+        if(mc.gameSettings.showDebugInfo)
+        {
+            final WorldClient world = mc.theWorld;
+            final EntityPlayerSP player = mc.thePlayer;
+            IColony colony = ColonyManager.getIColony(world,player.getPosition());
+            final double minDistance = ColonyManager.getMinimumDistanceBetweenTownHalls();
+
+            if(colony == null)
+            {
+                colony = ColonyManager.getClosestIColony(world, player.getPosition());
+
+                if (colony == null || Math.sqrt(colony.getDistanceSquared(player.getPosition())) > 2 * minDistance)
+                {
+                    event.getLeft().add(LanguageHandler.format("com.minecolonies.gui.debugScreen.noCloseColony"));
```

Comment:
We don't need the mindistance text if we're able to place our colony right away.